transcom/mymove
# Update pdfcpu to current version (and do not use fork)

## Description
We need to update the pdfcpu library in order to incorporate changes that resolve an issue that we were experiencing (see [pivotal story](https://www.pivotaltracker.com/n/projects/2181745/stories/166705097)). 

We were previously using / maintaining a fork of [pdfcpu](https://github.com/trussworks/pdfcpu), b/c `pdfcpu` required reading from files instead of accepting an interface like `io.Reader`. It's been a while since the fork was merged with the upstream repo (~ 103 commits behind upstream master), so merging in the new changes is nontrivial. However, pdfcpu now allows reading from anything that [supports the `io.ReadSeeker` interface](https://github.com/hhrutter/pdfcpu/blob/master/pkg/pdfcpu/read.go#L57), which I think addresses why we created a fork initially. So this PR reverts back to using the pdfcpu library in lieu of our fork. 

Also, this fixes two other issues noticed when testing the updated pdf library: 
* The response shape was different than what the front end was expecting, so the user wasn't being routed to the newly generated pdf dceb94a0a98bc9a1a39d0cae98828e87c4e43c87. 
* The handler would return an error if a service member had orders, but did not have any other uploads. Updated this to allow the return of orders only 0cd813d826e3e2357ec1cdfc783f266f8ba6c33a.

## Setup

```sh
make server_test
```
1) Make a new service member and create new ppm (when testing this I noticed that uploads don't get mapped to the files by the migration (i.e. e2e migrations), so in order to capture the file you need to upload it via the uploader instead of via the migration).
2) Login to the office app and find your ppm. You should be able to download the various documents by clicking "Download Orders and Weight Tickets (PDF)" or "Download All Attachments (PDF)"

## Code Review Verification Steps


## References

* [Pivotal story](https://www.pivotaltracker.com/n/projects/2181745/stories/166705097) and [Pivotal story](https://www.pivotaltracker.com/n/projects/2181745/stories/166705947) for this change
